### PR TITLE
Add Julia support

### PR DESCRIPTION
**Pull request by tylerjthomas9**
I converted the algorithm to Julia for a project that I am working on. Created this PR incase you are interested in expanding the language support.

### CODE REVIEW

File: julia/src/BidAsk.jl [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+    n1 = nanmean(o .== h)
+    n2 = nanmean(o .== l)
+    n3 = nanmean(c1 .== h1)
+    n4 = nanmean(c1 .== l1)
+    n5 = nanmean(h .== l .&& l .== c1)
+
+    s2 = -4 * (w1 * e1 + w2 * e2) / ((1 - k * (n1 + n2) / 2) + (1 - n5) * (1 - k * (n3 + n4) / 2))
+
+    return s2
```
Comment (eguidotti) [inline julia/src/BidAsk.jl]: This should be
```julia
return sqrt(max(0, s2))
```

File: julia/test/runtests.jl [hunk 2]
```diff
@@ -0,0 +1,10 @@
+using BidAsk
+using CSV
+using DataFrames
+using Test
+
+@testset "edge()" begin
+    df = CSV.File("../../pseudocode/ohlc.csv")
+    estimate = edge(df.:Open, df.:High, df.:Low, df.:Close)
+    @test isapprox(0.00010101064175748407, estimate)
```
Comment (eguidotti) [inline julia/test/runtests.jl]: This should test
```julia
@test isapprox(0.0100504050543988, estimate)
```

Comment (eguidotti) [inline julia/test/runtests.jl:9]: This should test
```julia
@test isapprox(0.0100504050543988, estimate)
```

**Comment by eguidotti** [comment]
Thanks a lot, @tylerjthomas9! I'm definitely interested in expanding the language support. This PR looks great! 

It seems you are returning the squared spread instead of the spread. What's the reason for this choice? Would you be able to fix that as I suggest in the comments above?

**Comment by tylerjthomas9** [comment]
>Thanks a lot, @tylerjthomas9! I'm definitely interested in expanding the language support. This PR looks great!

Thank you for putting together the repo. It was extremely useful.

> It seems you are returning the squared spread instead of the spread. What's the reason for this choice? Would you be able to fix that as I suggest in the comments above?

Oops, this was not intentional. I have made the change.

**Comment by eguidotti** [comment]
Thank you for the quick fix! 

I can see that the test is succesful. But it raises an error when missing values are present in the price data.
```julia
edge(): Error During Test at REPL[9]:1
  Got exception outside of a @test
  TypeError: non-boolean (Missing) used in boolean context
  Stacktrace:
    [1] is_nan_inf(x::Missing)
      @ BidAsk ~/.julia/packages/BidAsk/IA06B/src/BidAsk.jl:6
```

I'm not familiar with Julia, but it seems [`skipmissing`](https://docs.julialang.org/en/v1/manual/missing/) should be used instead of `is_nan_inf` to filter out missing values. Maybe something like `mean(skipmissing(x))`? Same for the variance.

**Comment by tylerjthomas9** [comment]
> I'm not familiar with Julia, but it seems [`skipmissing`](https://docs.julialang.org/en/v1/manual/missing/) should be used instead of `is_nan_inf` to filter out missing values. Maybe something like `mean(skipmissing(x))`? Same for the variance.

Good catch, I was looking to skip `inf` values too, but I messed up and didn't skip missing values. Modified the code to just skip missing values, so it is more consistent with the other implementations.

**Comment by eguidotti** [comment]
I replaced the logical `&&` operator with the bitwise `&` operator to support missing values (see https://discourse.julialang.org/t/error-typeerror-non-boolean-missing-used-in-boolean-context/83152)

This PR seems ready to merge. Thanks a lot for your contribution!

**Pull request merged**